aim/agent/aid/universes/aci/tenant.py

```diff
@@ -618,22 +624,25 @@ def _push_aim_resources(self):
                                     to_push)
                                 LOG.debug("POSTING into APIC: %s" % to_push)
                                 self._post_with_transaction(to_push)
-                                self.creation_succeeded(aim_object)
                             else:
-                                to_delete, to_update = (
+                                to_delete, to_delete_tag, to_update = (
                                     self.ownership_mgr.set_ownership_change(
                                         to_push))
-                                LOG.debug("DELETING from APIC: %s" % to_delete)
-                                for obj in to_delete:
+                                LOG.debug("DELETING tags from APIC: "
+                                          "%s" % to_delete_tag)
+                                # Delete object ownership
+                                for obj in to_delete_tag:
                                     attr = list(obj.values())[0]['attributes']
                                     self.ac_context.aci_session.DELETE(
                                         '/mo/%s.json' % attr.pop('dn'))
+                                LOG.debug("DELETING from APIC: %s" % to_delete)
+                                self._post_with_transaction(to_delete,
+                                                            deleted=True)
                                 LOG.debug("UPDATING in APIC: %s" % to_update)
                                 # Update object ownership
                                 self._post_with_transaction(to_update,
                                                             modified=True)
-                                if to_update:
```

Comment:
I'm not sure I follow the logic here to make this conditional always happen. Before, it would only happen if to_update was true?

Reply:
Before it would happen for create and update. It would be good for testing purposes to use it for delete as well.